tests/framework/extensions/hardening/k3s/harden_nodes.go

```diff
@@ -50,36 +51,53 @@ func HardeningNodes(client *rancher.Client, hardened bool, nodes []*nodes.Node,
 			if err != nil {
 				return err
 			}
-			err = node.SCPFileToNode(dirPath+"/psp.yaml", "/home/"+node.SSHUser+"/psp.yaml")
-			if err != nil {
-				return err
-			}
-			err = node.SCPFileToNode(dirPath+"/system-policy.yaml", "/home/"+node.SSHUser+"/system-policy.yaml")
-			if err != nil {
-				return err
-			}
 
-			logrus.Infof("Applying hardened YAML files to node: %s", node.NodeID)
 			_, err = node.ExecuteCommand("sudo bash -c 'mv /home/" + node.SSHUser + "/audit.yaml /var/lib/rancher/k3s/server/audit.yaml'")
 			if err != nil {
 				return err
 			}
-			_, err = node.ExecuteCommand("sudo bash -c 'mv /home/" + node.SSHUser + "/psp.yaml /var/lib/rancher/k3s/psp.yaml'")
-			if err != nil {
-				return err
-			}
-			_, err = node.ExecuteCommand("sudo bash -c 'mv /home/" + node.SSHUser + "/system-policy.yaml /var/lib/rancher/k3s/system-policy.yaml'")
-			if err != nil {
-				return err
-			}
 
-			_, err = node.ExecuteCommand("sudo bash -c 'export KUBECONFIG=/etc/rancher/k3s/k3s.yaml && kubectl apply -f /var/lib/rancher/k3s/psp.yaml'")
-			if err != nil {
-				return err
-			}
-			_, err = node.ExecuteCommand("sudo bash -c 'export KUBECONFIG=/etc/rancher/k3s/k3s.yaml && kubectl apply -f /var/lib/rancher/k3s/system-policy.yaml'")
-			if err != nil {
-				return err
+			if kubeVersion <= string(provisioninginput.PSPKubeVersionLimit) {
+				err = node.SCPFileToNode(dirPath+"/psp.yaml", "/home/"+node.SSHUser+"/psp.yaml")
+				if err != nil {
+					return err
+				}
+
+				err = node.SCPFileToNode(dirPath+"/system-policy.yaml", "/home/"+node.SSHUser+"/system-policy.yaml")
+				if err != nil {
+					return err
+				}
+
+				logrus.Infof("Applying hardened YAML files to node: %s", node.NodeID)
+				_, err = node.ExecuteCommand("sudo bash -c 'mv /home/" + node.SSHUser + "/psp.yaml /var/lib/rancher/k3s/psp.yaml'")
+				if err != nil {
+					return err
+				}
+
+				_, err = node.ExecuteCommand("sudo bash -c 'mv /home/" + node.SSHUser + "/system-policy.yaml /var/lib/rancher/k3s/system-policy.yaml'")
+				if err != nil {
+					return err
+				}
+
+				_, err = node.ExecuteCommand("sudo bash -c 'export KUBECONFIG=/etc/rancher/k3s/k3s.yaml && kubectl apply -f /var/lib/rancher/k3s/psp.yaml'")
+				if err != nil {
+					return err
+				}
+
+				_, err = node.ExecuteCommand("sudo bash -c 'export KUBECONFIG=/etc/rancher/k3s/k3s.yaml && kubectl apply -f /var/lib/rancher/k3s/system-policy.yaml'")
+				if err != nil {
+					return err
+				}
+			} else {
+				err = node.SCPFileToNode(dirPath+"/psa.yaml", "/home/"+node.SSHUser+"/psa.yaml")
+				if err != nil {
+					return err
+				}
+
+				_, err = node.ExecuteCommand("sudo bash -c 'mv /home/" + node.SSHUser + "/psa.yaml /var/lib/rancher/k3s/server/psa.yaml'")
+				if err != nil {
+					return err
+				}
```

Comment:
Would it make sense to wrap this as another function and call it here? Would a standalone function for the below version be beneficial?

Reply:
Personally I'd want to leave as is. I know it's a bit lengthy, this is all part of hardening the nodes so I think it makes sense to leave it all in once function.